### PR TITLE
[llvm] Fix passing character to subroutine with bindC convention

### PR DESCRIPTION
**Pull request by serge-sans-paille**
Under bindC conventions, character types are passed as byte stream.
Under normal conventions, they are represented as a pointer to a byte stream.
    
As a consequence, when calling a bindC function, they need to be
referenced through a load when packing the arguments, and their address
needs to be taken when unpacking them.

### CODE REVIEW

File: integration_tests/char_passing.f90 [hunk 1]
```diff
@@ -0,0 +1,13 @@
+program char_passing
```
Comment (Pranavchiku) [inline integration_tests/char_passing.f90:1]:
```suggestion
program character_05
```

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
You also need to register it inside `integration_tests/CMakeLists.txt`, you can apply following diff. Also I'll ask you to rename your test file to `character_05.f90` just to be consistent and group it with other tests.

```diff
diff --git a/integration_tests/CMakeLists.txt b/integration_tests/CMakeLists.txt
index 05ace67ac..e1365b13e 100644
--- a/integration_tests/CMakeLists.txt
+++ b/integration_tests/CMakeLists.txt
@@ -1478,6 +1478,7 @@ RUN(NAME character_01 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc)
 RUN(NAME character_02 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc)
 RUN(NAME character_03 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc)
 RUN(NAME character_04 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc)
+RUN(NAME character_05 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc)
 RUN(NAME char_array_01 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc)
 RUN(NAME char_array_indexing LABELS gfortran llvm llvm_wasm llvm_wasm_emcc)
 RUN(NAME character_parameter_padding_trimming LABELS gfortran llvm llvm_wasm llvm_wasm_emcc)
```

**Review by Pranavchiku — APPROVED** [review]
This is good, I'll enable auto-merge. One question, can we pass in `character(len=5) :: b`?

**Comment by serge-sans-paille** [comment]
> This is good, I'll enable auto-merge. One question, can we pass in `character(len=5) :: b`?

Just tested, and yes

**Comment by Pranavchiku** [comment]
let's then add a test for it.

**Review by Pranavchiku — COMMENTED** [review]
I think you have not added `character_06.f90` file

**Comment by certik** [comment]
This looks like the right fix, but I think we only want to load it sometimes, given that some other test now fails with this change. We have to figure out the condition when to load it.

**Comment by serge-sans-paille** [comment]
@Pranavchiku / @certik nailed it :-)

**Review by Pranavchiku — APPROVED** [review]
looks good to me, I'll wait for certik to have a look.

**Review by certik — APPROVED** [review]
Awesome, thanks for the fix!

**Comment by certik** [comment]
This is another instance where https://github.com/lfortran/lfortran/issues/4245 would help.

**Pull request merged**